### PR TITLE
no-duplicate-dollar-variables: allow dup vars in unrelated scopes

### LINKED ISSUE #277: no-duplicate-dollar-variables doesn't respect scopes
`no-duplicate-dollar-variables` rule reports variables with the same name but declared in different scopes. The snippet below produces error `Unexpected duplicate dollar variable $padding-x   scss/no-duplicate-dollar-variables`.

```
.FormControl-label {
  $padding-x: $padding-small-x;

  @include input-overlay;

  position: relative;
  top: 0.40em;
  margin-left: $margin-medium-x;
  padding-left: $padding-x;
  padding-right: $padding-x;
  font-size: $font-size-small;
}


.FormControl-input {
  $padding-x: $padding-medium-x;

  width: 100%;
  border: 1px solid $color-grey-light;
  border-radius: 0;
  padding-left: $padding-x;
  padding-right: $padding-x;
  height: $row-height;
}
```

For me it's pretty obvious that this shouldn't be reported. Am I wrong?

### PR DESCRIPTION
**Pull request by srawlins**
Fixes #277

### CODE REVIEW

File: src/rules/no-duplicate-dollar-variables/index.js [hunk 1]
```diff
@@ -60,7 +95,19 @@ export default function(value, secondaryOptions) {
         return;
       }
 
-      if (vars[decl.prop]) {
+      const ancestors = [];
+      let parent = decl.parent;
+
+      while (parent !== null && parent !== undefined) {
+        const parentKey = parent.toString();
```
Comment (srawlins) [inline src/rules/no-duplicate-dollar-variables/index.js]: This seems to be inefficient - I think this serializes the entire subtree below [parent]. I'm not sure what to use as unique keys to traverse down the scopes though. the non-leaf nodes are all [Container](https://api.postcss.org/Container.html)s I think? I couldn't find a good common way to uniquify them...

**Comment by kristerkari** [comment]
Thanks! This seems like a good fix to the rule.

**Comment by srawlins** [comment]
Done. Additional test and README text.

**Comment by kristerkari** [comment]
Great, thanks! Going to do a new release with the currently approved PRs.

**Pull request merged**